pyathena/common.py

```diff
@@ -142,24 +142,64 @@ def _build_start_query_execution_request(self, query, work_group=None, s3_stagin
             })
         return request
 
-    def _execute(self, operation, parameters=None, work_group=None, s3_staging_dir=None):
+    def _find_previous_query_id(self, query, work_group, cache_size):
```

Comment:
This method may return a QueryID or nothing. If possible, I'd like to return QueryID or None explicitly.
For example, the following implementation:
```
    def _find_previous_query_id(self, query, work_group, cache_size):
        query_id = None
        try:
            next_token = None
            while cache_size > 0:
                n = min(cache_size, 50)  # 50 is max allowed by AWS API
                cache_size -= n
                request = {'MaxResults': n}
                if work_group is not None:
                    request['WorkGroup'] = work_group
                if next_token is not None:
                    request['NextToken'] = next_token
                response = retry_api_call(self.connection._client.list_query_executions,
                                          config=self._retry_config,
                                          logger=_logger,
                                          **request)
                query_ids = response.get('QueryExecutionIds', None)
                if not query_ids:
                    break  # no queries left to check
                next_token = response.get('NextToken', None)
                query_executions = retry_api_call(
                    self.connection._client.batch_get_query_execution,
                    config=self._retry_config,
                    logger=_logger,
                    QueryExecutionIds=query_ids
                ).get('QueryExecutions', [])
                for execution in query_executions:
                    if execution['Query'] == query and \
                            execution['Status']['State'] == AthenaQueryExecution.STATE_SUCCEEDED and \
                            execution['StatementType'] == 'DML':
                        query_id = execution['QueryExecutionId']
                        break
                if query_id or next_token is None:
                    break
        except Exception:
            _logger.warning('Failed to check the cache. Moving on without cache.')
        return query_id
```

Reply:
Won't this fail your linting which restricts lines to <= 100 characters?

Reply:
Yep, it fails the linting. https://travis-ci.org/laughingman7743/PyAthena/jobs/596109171#L289